RecoTracker/TkSeedGenerator/plugins/DeepCoreSeedGenerator.cc

```diff
@@ -0,0 +1,583 @@
+// -*- C++ -*-
+//
+// Package:    trackJet/DeepCoreSeedGenerator
+// Class:      DeepCoreSeedGenerator
+//
+/**\class DeepCoreSeedGenerator DeepCoreSeedGenerator.cc trackJet/DeepCoreSeedGenerator/plugins/DeepCoreSeedGenerator.cc
+
+ Description: [one line class summary]
+
+ Implementation:
+     [Notes on implementation]
+*/
+//
+// Original Author:  Valerio Bertacchi
+//         Created:  Mon, 18 Dec 2017 16:35:04 GMT
+//
+//
+
+// system include files
+
+#include <memory>
+
+// user include files
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "Geometry/CommonDetUnit/interface/GlobalTrackingGeometry.h"
+#include "Geometry/Records/interface/GlobalTrackingGeometryRecord.h"
+#include "Geometry/CommonDetUnit/interface/GlobalTrackingGeometry.h"
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include "DataFormats/Common/interface/DetSet.h"
+#include "DataFormats/SiPixelCluster/interface/SiPixelCluster.h"
+#include "DataFormats/VertexReco/interface/Vertex.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/VertexReco/interface/VertexFwd.h"
+#include "DataFormats/JetReco/interface/Jet.h"
+#include "DataFormats/SiPixelDigi/interface/PixelDigi.h"
+#include "DataFormats/GeometryVector/interface/VectorUtil.h"
+#include "DataFormats/Math/interface/Point3D.h"
+#include "DataFormats/Math/interface/Vector3D.h"
+#include "DataFormats/Candidate/interface/Candidate.h"
+
+#include "RecoLocalTracker/ClusterParameterEstimator/interface/PixelClusterParameterEstimator.h"
+#include "RecoLocalTracker/Records/interface/TkPixelCPERecord.h"
+
+#include "TrackingTools/GeomPropagators/interface/StraightLinePlaneCrossing.h"
+#include "TrackingTools/GeomPropagators/interface/Propagator.h"
+#include "TrackingTools/Records/interface/TrackingComponentsRecord.h"
+
+#include "MagneticField/Engine/interface/MagneticField.h"
+#include "MagneticField/Records/interface/IdealMagneticFieldRecord.h"
+
+#include "FWCore/ServiceRegistry/interface/Service.h"
+
+#include "Geometry/CommonDetUnit/interface/PixelGeomDetUnit.h"
+
+#include "DataFormats/TrajectorySeed/interface/TrajectorySeedCollection.h"
+
+#include "PhysicsTools/TensorFlow/interface/TensorFlow.h"
+
+//
+// class declaration
+//
+struct DeepCoreCache {
+  mutable tensorflow::GraphDef* graph_def;
```

Comment:
#32271 that changes the `tensorflow::createSession()` to take `const GraphDef *` has been merged now so the `mutable` could be removed (after a rebase to a recent IB).

Reply:
right, thanks for pointing to it!

Reply:
thank you @makortel , what's the easier way to have this feature locally and be able to test it? Because if I simply change mutable to const in my branch it doesn't work, becsause I don't have the feature. A way can be to checkout the most updated release (after the merge of the feature), merge this PR on that new branch, test, and do another PR to master from that new branch... but in this way I have to close *this* PR. Other possibilities? Obviously if we are 100% sure that this is not breaking anything I can directly change and push also if it is not compiling locally. Let me know please what you prefer.

Reply:
You don't have to close this PR. You would need to check out the most recent IB and then do a rebase to master: https://cms-sw.github.io/tutorial-resolve-conflicts.html.
After the rebase, if everything looks correct, you can force push to the branch this PR is based on.

Reply:
thank you, I will follow this approch (I didn't know that a forced push like the one you suggest is allowed). Can I do the rebase after I finished the igprof test or do you think that would be better to profile the "const version" of the graph?

Reply:
AFAIK "force push" is the only way to update a PR after a rebase. Of course, you can only do this to your own branches, and you should be very careful (make a backup).

In principle the const version would be more appropriate as it might be slightly more efficient, but since these tests take time, let's have a look at the results you have so far with the current version if we can already make sure that the results make sense. We will only ask for another profiling if there are big questions still to be answered.

Reply:
ok thanks, I will post here the igprof asap, then I'll proceed with the rebase mutable->const fix.